src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyNameFormatter.cs

```diff
@@ -99,24 +99,32 @@ private static void AppendQuoted(this StringBuilder sb, string s)
 
             for (int i = 0; i < s.Length; i++)
             {
-                bool addedEscape = false;
-                foreach (KeyValuePair<char, string> kv in EscapeSequences)
+                switch (s[i])
                 {
-                    string escapeReplacement = kv.Value;
-                    if (!(s[i] == escapeReplacement[0]))
+                    case '\\':
+                    case ',':
+                    case '=':
+                    case '\'':
+                    case '"':
+                        sb.Append('\\');
+                        break;
+                    case '\t':
+                        sb.Append('\\').Append('t');
                         continue;
-                    if ((s.Length - i) < escapeReplacement.Length)
+                    case '\n':
+                        sb.Append('\\').Append('n');
                         continue;
-                    if (s.AsSpan(i, escapeReplacement.Length).SequenceEqual(escapeReplacement))
-                    {
-                        sb.Append('\\');
-                        sb.Append(kv.Key);
-                        addedEscape = true;
-                    }
+#if TARGET_WINDOWS
+                    case '\r':
+                        if (++i < s.Length - 1 && s[i] == '\n')
+                            goto case '\n';
+
+                        --i;
+                        break;
+#endif
```

Comment:
Is this a change in behavior?  Seems like a \n on Windows (no proceeding \r) would have still resulted in a \r\n, but now it's only going to be a \n?

Reply:
And previously \r wasn't special-cased if it was on its own, was it?

Reply:
Good point simplified it back